ext/cfscrape/__init__.py

```diff
@@ -152,75 +153,112 @@ def solve_cf_challenge(self, resp, **original_kwargs):
             raise ValueError(
                 "Unable to parse Cloudflare anti-bot IUAM page: %s %s"
-                % (e.message, BUG_REPORT)
+                % (e, BUG_REPORT)
             )
 
         # Solve the Javascript challenge
         answer, delay = self.solve_challenge(body, domain)
-        params["jschl_answer"] = answer
+        if method == 'POST':
+            cloudflare_kwargs["data"]["jschl_answer"] = answer
+        elif method == 'GET':
+            cloudflare_kwargs["params"]["jschl_answer"] = answer
 
         # Requests transforms any request into a GET after a redirect,
         # so the redirect has to be handled manually here to allow for
         # performing other types of requests even as the first request.
-        method = resp.request.method
         cloudflare_kwargs["allow_redirects"] = False
 
         # Cloudflare requires a delay before solving the challenge
         time.sleep(max(delay - (time.time() - start_time), 0))
 
         # Send the challenge response and handle the redirect manually
         redirect = self.request(method, submit_url, **cloudflare_kwargs)
-        redirect_location = urlparse(redirect.headers["Location"])
-
-        if not redirect_location.netloc:
-            redirect_url = urlunparse(
-                (
-                    parsed_url.scheme,
-                    domain,
-                    redirect_location.path,
-                    redirect_location.params,
-                    redirect_location.query,
-                    redirect_location.fragment,
+        if "Location" in redirect.headers:
+            redirect_location = urlparse(redirect.headers["Location"])
+
+            if not redirect_location.netloc:
+                redirect_url = urlunparse(
+                    (
+                        parsed_url.scheme,
+                        domain,
+                        redirect_location.path,
+                        redirect_location.params,
+                        redirect_location.query,
+                        redirect_location.fragment,
+                    )
                 )
-            )
-            return self.request(method, redirect_url, **original_kwargs)
-        return self.request(method, redirect.headers["Location"], **original_kwargs)
+                return self.request(method, redirect_url, **original_kwargs)
+            return self.request(method, redirect.headers["Location"], **original_kwargs)
+        elif 'cf_clearance' in redirect.headers['Set-Cookie']:
+            resp = self.request(self.org_method, submit_url, cookies = redirect.cookies)
+            return resp
+        else:
+            resp = self.request(self.org_method, submit_url, **cloudflare_kwargs)
+            return resp
+
 
     def solve_challenge(self, body, domain):
         try:
+            javascript = re.search(r'\<script type\=\"text\/javascript\"\>\n(.*?)\<\/script\>',body, flags=re.S).group(1) # find javascript
```

Comment:
@medariox 
Sorry for being so late, but didn't you also need to update the monkeypatch with the changes in this function?
https://github.com/pymedusa/Medusa/blob/develop/medusa/init/cfscrape.py

Reply:
Yeah, in theory I should have. I tested it and everything worked... so I forgot about it 😖 I guess we are fine...? 😕

Reply:
I don't use anything that has CF protection so I wouldn't know haha.
I guess you'll have to wait and see. 😜